src/main/java/seedu/bookmark/logic/parser/ParserUtil.java

```diff
@@ -89,4 +92,32 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses {@code String totalPages} into a {@code TotalPages}.
+     */
+    public static TotalPages parseTotalPages(String totalPages) throws ParseException {
+        requireNonNull(totalPages);
+        String trimmedTotalPages = totalPages.trim();
+        if (!TotalPages.isValidTotalPages(trimmedTotalPages)) {
+            throw new ParseException(TotalPages.MESSAGE_CONSTRAINTS);
+        }
+        return new TotalPages(trimmedTotalPages);
+    }
+
+    /**
+     * Parses {@code Optional<String> bookmark} into a {@code Bookmark}
+     */
+    public static Bookmark parseBookmark(Optional<String> bookmark, TotalPages totalPages) throws ParseException {
+        if (bookmark.equals(Optional.empty())) {
```

Comment:
Perhaps using `bookmark.isEmpty()` could replace `bookmark.equals(Optional.empty())`

Reply:
Good recommendation! Thank you for spotting this. I've made the relevant changes.